config/v1/types_authentication.go

```diff
@@ -249,8 +253,44 @@ type TokenIssuer struct {
 	// the "ca-bundle.crt" key.
 	// If unset, system trust is used instead.
 	CertificateAuthority ConfigMapNameReference `json:"issuerCertificateAuthority"`
+
+	// discoveryURL, if specified, overrides the URL used to fetch discovery
+	// information instead of using "{url}/.well-known/openid-configuration".
+	// The exact value specified is used, so "/.well-known/openid-configuration"
+	// must be included in discoveryURL if needed.
+	// Must use the https:// scheme.
+	//
+	// +kubebuilder:validation:Pattern=`^https:\/\/[^\s]`
+	// Ensure that discoveryURL is unique within the set of seenDiscoveryURLs
+	// +kubebuilder:validation:XValidation:rule="!seenDiscoveryURLs.has(self.issuer.discoveryURL)",message="discoveryURL must be unique"
+	// +kubebuilder:validation:XValidation:rule="self.issuer.discoveryURL.size() > 0 ? (self.issuer.url.size() == 0 || self.issuer.url.trim('/').notEquals(self.issuer.discoveryURL.trim('/')))" ,message="discoveryURL must be different from URL"
+	// +kubebuilder:validation:XValidation:rule="self.issuer.discoveryURL.size() > 0 ? has(self.StructuredAuthenticationConfiguration) && self.StructuredAuthenticationConfiguration" ,message="discoveryURL is not supported when StructuredAuthenticationConfiguration feature gate is disabled"
+
+	// +optional
+	DiscoveryURL string `json:"discoveryURL,omitempty"`
+
+	// AudienceMatchPolicy controls how the "aud" claim in JWT tokens is validated.
+	// It allows flexible matching of the audience value in tokens.
+	// Possible values: MatchAny, MatchAll.
+	// +kubebuilder:validation:XValidation:rule="self.audienceMatchPolicy.size() > 0",message="At least one audienceMatchPolicy is required"
+	// +kubebuilder:validation:XValidation:rule="self.audienceMatchPolicy.size() <= 1 || (has(self.StructuredAuthenticationConfiguration) && self.StructuredAuthenticationConfiguration)",message="Multiple audienceMatchPolicy values are not supported when StructuredAuthenticationConfiguration feature gate is disabled"
```

Comment:
As I mentioned in a previous comment, there is no way to check a feature gate in the CEL expression. Remove this validation.